### PR TITLE
Updated docs with native installation steps for Linux and OS X

### PR DESCRIPTION
**Pull request by chzbrgr71**
Updated the base README to offer both Docker and native (but Docker is recommended). Instructions updated with steps for Linux and fixed a couple items in the OS X steps. Tested with Ubuntu Server 16.04.

Did not change or test anything for the Windows portion.

### CODE REVIEW

**Comment by colemickens** [comment]
Can you add a (slash move the existing) Docker section to the dev guide? The changes here look great other than that bit of info that is being lost. It's still nice to have documented for those that don't want to have to setup GOPATH or install tooling into their local dev box.

**Comment by chzbrgr71** [comment]
By dev guide, you mean 'acsengine.md' right? 

Totally agree about the lost info. To me, it looked like there are 3 experiences here. One is running in Docker. Two is installing/running natively. And third is for the developer that wants to update acs-engine code itself in a developer environment. I imagine that third area could easily be added if it isn't clearly covered with the above.

**Comment by colemickens** [comment]
You don't need to close it, you can force push another commit and we can get this merged.

I don't understand the third scenario there. Isn't that just "working with Git(Hub) 101"? I don't really want to document every single thing under the sun for Git, Go, etc. There's plenty of other resources out there; we can link to them if appropriate.

**Comment by chzbrgr71** [comment]
Agreed. Update coming shortly.

File: README.md [hunk 1]
```diff
@@ -26,45 +26,6 @@ The cluster definition file enables the following customizations to your Docker
 * [Managed Disks](examples/disks-managed) (under private preview) - shows how to use managed disks 
 * [Large Clusters](examples/largeclusters) - shows how to create cluster sizes of up to 1200 nodes
 
-## Development (Docker)
```
Comment (anhowe) [inline README.md]: Please ensure you update all documents that point to this section.  I would recommend instead leave "Development (Docker)" here and point to the new location.

Comment (anhowe) [inline README.md]: I thought this was linked in a few more places, but I think you found the only link in you change below so this is good.

**Comment by colemickens** [comment]
Thanks for the contribution @chzbrgr71 !

**Pull request merged**